### PR TITLE
add correctionlib and dependencies. 

### PR DESCRIPTION
**Pull request by davidlange6**
Also adds python-rapidjson and pydantics as dependencies.

Currently pip installs only the python interface. There is a C++ one as well that hopefully we get from a future pypi release

@nsmith-

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: pip/requirements.txt [hunk 1]
```diff
@@ -251,6 +253,7 @@ python-cjson==1.2.2;python_version<'3.0'
 python-daemon==2.2.4
 python-dateutil==2.8.1
 python-ldap==3.3.1
+python-rapidjson==1.0;python_version>'3.0'
```
Comment (nsmith-) [inline pip/requirements.txt:256]: Does this need to be there to build? The rapidjson headers should have been shipped with the correctionlib sdist, and we don't use rapidjson on the python side at all.

**Comment by davidlange6** [comment]
Possibly not. Seemed otherwise useful to have.

**Comment by smuzaffar** [comment]
@davidlange6 , please add `py3-correctionlib` to python_tools ? Can you please also add simple unit tests in cmssw for these new tools?

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals

**Pull request merged**